src/frontend/src/lib/utils/tokens.utils.ts

```diff
@@ -49,6 +49,36 @@ export const sortTokens = ({
 	];
 };
 
+/**
+ * Calculates USD balance for the provided token.
+ *
+ * @param token - Token for which USD balance will be calculated.
+ * @param $balancesStore - The balances data for the tokens.
+ * @param $exchanges - The exchange rates data for the tokens.
+ * @returns The USD balance or undefined in case the number cannot be calculated.
+ *
+ */
+export const calculateTokenUsdBalance = ({
+	token,
+	$balances,
+	$exchanges
+}: {
+	token: Token;
+	$balances: CertifiedStoreData<BalancesData>;
+	$exchanges: ExchangesData;
+}): number | undefined => {
+	const balance: BigNumber | undefined = $balances?.[token.id]?.data;
+	const exchangeRate: number | undefined = $exchanges?.[token.id]?.usd;
+
+	return nonNullish(exchangeRate)
+		? usdValue({
+				token,
+				balance,
```

Comment:
maybe you don't need to create the `balance` as `const` and pass it here directly?

Reply:
done